xds/core/v3/extension.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package xds.core.v3;
+
+option java_outer_classname = "ExtensionProto";
+option java_multiple_files = true;
+option java_package = "com.github.udpa.xds.core.v1";
```

Comment:
It would break the generated Java code, so we're extending API compatibility to generated code I don't think we can change it?

Fixed the version string

Reply:
But this is lift-and-shift across repos, so we don't have strict compatibility requirements, for API or code?

Reply:
Sure we could change the package of this new file to be com.github.xds.core.v1, but won't that make this inconsistent with the other packages in here? I was thinking about the idea of changing the java_package of all the protos in xds/ to not include udpa, which affects more than just this file

Reply:
I think we should do the following:
1. Decide what we want the java package path to be for all of the xds protos.
2. Use the new path for these new files.
3. Decide on a plan to fix the existing files.  If we can just change them without breaking anything, we should do that; otherwise, we need to come up with an alternative deprecate-and-migrate plan.

@ejona86 may have some thoughts on how this affects things for Java.

Reply:
I think let's go with `com.github.xds.core.v3` and make sure everything in at least the `xds/` tree is consistent.

Reply:
Can we go ahead and change the existing files in the xds tree?  Or will that break existing usage?

Reply:
What is the migration process expected for this proto? It has a different package name and file path, so it seems like a brand new proto from protobuf's perspective. In that case, it is a new proto with no prior usages and the Java path can be anything but `io.envoyproxy.envoy.config.core.v3` (to avoid colliding with the other copy).

Reply:
Mark spoke to me and made clear the issue is the existing files.

The existing files are [already included in grpc-java](
https://github.com/grpc/grpc-java/tree/master/xds/third_party/udpa/src/main/proto/xds/core/v3), however, [there aren't any references to the messages](https://github.com/grpc/grpc-java/search?q=com.github.udpa.xds.core.v3) outside of generated code. That means that as long as the consuming protos are generated in the same build process as these protos we'd be fine. That is true for Google-internal and, since grpc-java is not using packages from maven-central but instead building and shading the generated code in grpc-xds. it would be true for the github builds of grpc-java as well.

So it seems fine to change these package names from grpc-java's perspective.

Reply:
Okay, super, I'll send a separate PR to update the existing files.

Should we use `com.github.cncf.xds` instead of just `com.github.xds`, to avoid potential conflicts with some other github `<something>/xds` repo?

Reply:
Actually, nevermind that last question -- we're already using `xds` as the top-level proto package, so `com.github.xds` is probably fine.

Reply:
Sent #9 with the changes to the existing files.